README.md

```diff
@@ -2,41 +2,72 @@
 
-The wrapper script has the following general structure
+The wrapper script is called by the cluster job that was submitted by Roddy in a fire-and-forget fashion:
+
+![image](images/sequence.svg)
+
+The wrapper script does the following steps
+
+![image](images/state.svg)
+
+## Calling Conventions of the `wrapInScript.sh`
+
+The `wrapInScript.sh` is called without any parameters. 
+The entire configuration of the `wrapInScript.sh` and the wrapped top-level bioinformatic script is done via environment variables and a configuration file.
+
+* `PARAMETER_FILE` is the path to the parameter file that is loaded (and thus evaluated) by the Bash `source` command.
+The parameter file should `export` all variables and `export -f` all functions that are needed by the wrapped script.
+* `debugWrapInScript` is a string variable that turns on debugging options in the `wrapInScript.sh`.
+* `outputFileGroup`: Can be set to the name of existing group of which the used user is a member. This group will be used for the output files of the wrapped script.
+* `sgWasCalled`: Should be set to "false" to prevent `sg` from being called to change the primary group (the group used for creating files) before the wrapped script is called.
+
+## Files created
 
-  - setup
-  - source `baseEnvironmentScript` (e.g. `/etc/profile`)
-  - source the job parameter file (`PARAMETER_FILE`)
-  - optionally, if `outputFileGroup` != "false", change to the requested group with `sg` and restart the script at the "setup" step (above)
-  - source job-specific environment script (see "Environment Setup Support" below)
-  - setup scratch directory
-  - update the `jobStateLogfile.txt` using a lock-file
-  - run the wrapped script using bash 
-  - kill child-process still running after the wrapped script ended
-  - update the `jobStateLogfile.txt` with the job's exit code
-  - exit
+* job-state logfile:
+* job-state logfile lock-file: This has the same name and location as the job-state logfile, but with the suffix `~` appended.
+The lock is created only twice, before the job-state logfile is accessed, and deleted after each write operation was finished.
+* `extendedLogs` directory: This directory contains the extended logs of the wrapped script and creates a file for each cluster job (using the job's parameter file name as template).
+It contains two dumps of the environment during the execution of the wrapped script -- before and after the parameter file was `source`d.
 
 ## Environment Setup Support
 
 ### Base Environment Script
 
-Each job is started with the default environment configured in you `applicationProperties.ini` in the `baseEnvironmentScript` variable. The `baseEnvironmentScript` serves as kind of general configuration of your cluster environment. Usually you will use a script like `/etc/profile` or `$HOME/.profile` or `$HOME/.bashrc`. 
+Each job is started with the default environment configured in your `applicationProperties.ini` in the `baseEnvironmentScript` variable.
```

Comment:
in the `baseEnvironmentScript` variable of your `applicationProperties.ini` file.